test/KrakenIntegrationTest.cpp

```diff
@@ -180,7 +180,163 @@ TEST_F(KrakenIntegrationTest, VerifyCommunication) {
    auto endExpected = MergeData(session, SendType::End, noChunk, noError);
    EXPECT_TRUE((endExpected == vec)) << vectorToString(vec);
 
-   
+
 }
 
 
+
+
+// This integration test is imporant also since it's the only one that truly tests
+// the KrakenBattle::SendChunks (cpp hidden file)
+//
+// To verify the expected steps in this communication
+// 0. some header (in this case just 'hello')
+// 1. uuid<DATA>w,o,r,l,d
+// 2. uuid<DATA> - part1 10MB
+// ---> ABORT sent from the receiver/Harpoon
+// 3. uuid<DATA> - part2 10MB
+// 4. uuid<DONE>
+// 5. uuid<END>
+TEST_F(KrakenIntegrationTest, AbortedCommunication) {
+   using namespace KrakenBattle;
+   ASSERT_EQ(zctx_interrupted, false);
+
+
+
+   Kraken::Chunks header = {'h', 'e', 'l', 'l', 'o'};
+
+// 10MB
+   const size_t kMaxChunkSize_10MB = 10 * 1024 * 1024;
+   const std::string session = "some-random-session";
+   const auto chunk1 = Kraken::Chunks{'w', 'o', 'r', 'l', 'd'};
+   const auto chunk2 = GetRandomData(kMaxChunkSize_10MB);
+
+
+
+   const std::string queue = "tcp://127.0.0.1:15123";
+   Kraken kraken;
+   kraken.MaxWaitInMs(1000);
+   auto spear = kraken.SetLocation(queue);
+   EXPECT_EQ(spear, Kraken::Spear::IMPALED) << "spear: " << static_cast<int>(spear);
+
+   std::shared_ptr<std::atomic<size_t>> expectedToReceive {new std::atomic<size_t>{0}};
+   std::shared_ptr<Harpoon> harpoon = std::make_shared<Harpoon>();
+   auto spearResult = harpoon->Aim(queue);
+   ASSERT_EQ(spearResult, Harpoon::Spear::IMPALED);
+   harpoon->MaxWaitInMs(1000);
+
+   // RECEIVER
+   using HarpoonReceived = std::vector<Kraken::Chunks>;
+   std::future<HarpoonReceived> willReceive = std::async(std::launch::async,  [expectedToReceive, harpoon] {
+      StopWatch stopWatch;
+      HarpoonReceived data;
+      size_t counter = 0;
+      size_t abortAt = 3; // i.e. before the 4th we abort
+      while (expectedToReceive->load() == 0 && stopWatch.ElapsedSec() < 10) {
+         std::this_thread::sleep_for(std::chrono::milliseconds(100));
+      }
+
+
+      size_t expected = expectedToReceive->load();
+      while (data.size() != expected  && stopWatch.ElapsedSec() < 10 && !zctx_interrupted) {
+         auto harpoonResult = Harpoon::Battling::CONTINUE;
+         if (abortAt == counter) {
```

Comment:
Set another flag in this if block. Your test should expect that this code actually ran.